NeoOnnx/src/Optimizer.h

```diff
@@ -0,0 +1,62 @@
+/* Copyright © 2017-2020 ABBYY Production LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+--------------------------------------------------------------------------------------------------------------*/
+
+#pragma once
+
+#include "common.h"
+#include <NeoOnnx/NeoOnnxImport.h>
+
+namespace NeoOnnx {
+
+class IOptimizer : public IObject {
+public:
+	explicit IOptimizer( CDnn& graph, const char* const classesOfSkipLayers[] ) : Graph( graph ), ClassesOfSkipLayers( classesOfSkipLayers ) {}
+	IOptimizer( IOptimizer&& ) = delete;
+	IOptimizer( const IOptimizer& ) = delete;
+
+	virtual void Apply() = 0;
+
+protected:
+	CDnn& Graph;
+	const char * const * ClassesOfSkipLayers{};
+
+	// Operations on 'layersSelected' array
+	auto GetLayerSelectedSize() const { return layersSelected.Size(); }
```

Comment:
_very minor, may be ignored_
Inconsistent naming GetLayerSelectedSize vs AddToLayer!s!Selected and ClearLayer!s!Selected.

P.S. something like `SelectLayer`, `GetSelectedLayer`, `ClearLayerSelection` `LayerSelectionSize` sounds better in English.

Reply:
agree, corrected